### DISCUSSION TITLE
Retrieve workflow id

### DISCUSSION THREAD

**Discussion by akifarhan** [codemagic-ci-cd/codemagic-docs#2299] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hi Codemagic,

I'm working on integrating Discord notifications into my Codemagic workflows and would like to include the workflow ID that triggered the build in the message. I've attempted to use the `CM_WORKFLOW_ID` environment variable, but it doesn't seem to be working as expected.

Could you please advise on:

**Retrieving the workflow ID**: Is there a specific way to correctly extract the workflow ID within the post-build script?
**Available information**: Could you provide a list of environment variables or other accessible information that can be used in Discord messages?

### Expected results

Workflow ID: `prod-release`

### Actual results

Workflow ID:

**Comment by dtrdic** [comment, marked as answer]
Already been resolved on Discord but just to respond in here as well.

You can use the built in environment variable `CM_BUILD_ID` which is a UUID of the build.
Check out the rest of built in environment variables on this link https://docs.codemagic.io/yaml-basic-configuration/environment-variables/
